src/platform/backends/qemu/qemu_virtual_machine_factory.cpp

```diff
@@ -18,22 +18,47 @@
 #include "qemu_virtual_machine_factory.h"
 #include "qemu_virtual_machine.h"
 
+#include <multipass/cloud_init_iso.h>
 #include <multipass/format.h>
 #include <multipass/logging/log.h>
 #include <multipass/platform.h>
 #include <multipass/process/simple_process_spec.h>
 #include <multipass/virtual_machine_description.h>
+#include <multipass/vm_specs.h>
+#include <multipass/yaml_node_utils.h>
 
 #include <shared/qemu_img_utils/qemu_img_utils.h>
 
 #include <QRegularExpression>
 
 namespace mp = multipass;
 namespace mpl = multipass::logging;
+namespace mpu = multipass::utils;
 
 namespace
 {
 constexpr auto category = "qemu factory";
+
+namespace fs = std::filesystem;
+void copy_instance_dir_without_snapshot_files(const fs::path& source_instance_dir_path,
+                                              const fs::path& dest_instance_dir_path)
+{
+    if (fs::exists(source_instance_dir_path) && fs::is_directory(source_instance_dir_path))
+    {
+        for (const auto& entry : fs::directory_iterator(source_instance_dir_path))
+        {
+            fs::create_directory(dest_instance_dir_path);
+
+            const fs::path filename = entry.path().filename();
```

Comment:
`path filename() const;` returns a value, meaning a temporary object here. So it should be caught by value and utilize the return value optimization. However, const reference would still compile here because C++ deliberately specifies that binding a temporary object to a const reference and that will prolong the lifetime of the temporary object. 

I would say, the const reference to temporary object is more confusing in terms of expressing what it should do and might not take advantage of the return value optimization.